backends/arm/operator_support/tosa_supported_operators.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2024 Arm Limited and/or its affiliates.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# pyre-unsafe
+
+import operator
+
+import torch.fx as fx
+from executorch.backends.arm.tosa_specification import TosaSpecification
+from executorch.exir.dialects._ops import ops as exir_ops
+from torch.fx.passes.operator_support import OperatorSupportBase
+
+
+class SupportedTOSAOperatorCheck:
+    """
+    Supported OP for TOSA lowering
+    """
+
+    # Should be populated by subclass implementation
+    tosa_specs: list[TosaSpecification] = []
+    targets: list[str] = []
+
+    def is_node_supported(self, node: fx.Node, tosa_spec: TosaSpecification) -> bool:
+        """
+        Checks if the fx.Node node is lowerable using the TOSA specification defined by tosa_spec.
+        To be implemented by subclasses targeting
+        """
+        raise NotImplementedError("NodeVisitor must be extended.")
+
+
+# container for all SupportedTosaOperatorCheck classes
+_tosa_spec_dicts: dict[TosaSpecification, dict[str, SupportedTOSAOperatorCheck]] = {
+    TosaSpecification.create_from_string("TOSA-0.80.0+BI"): {},
+    TosaSpecification.create_from_string("TOSA-0.80.0+MI"): {},
+}
+
+
+def register_tosa_support_check(checker):
+    """
+    Decorator to mark a subclass implmentation of SupportedTosaOperatorCheck
+    to be registered for checking if a torch.fx.Node is lowerable given
+    a TOSA specification.
+    """
+    for tosa_spec in checker.tosa_specs:
+        for target in checker.targets:
+            _tosa_spec_dicts[tosa_spec][target] = checker
+    return checker
+
+
+def get_registered_tosa_support_checks(
+    tosa_spec: TosaSpecification,
+) -> dict[str, SupportedTOSAOperatorCheck]:
+
+    if tosa_spec not in _tosa_spec_dicts:
+        raise RuntimeError
+
+    tosa_support_checks = {}
+    for target, tosa_check in _tosa_spec_dicts[tosa_spec].items():
+        tosa_support_checks[target] = tosa_check()
```

Comment:
I think this should be:
```
tosa_support_checks[target] = tosa_check()
```
--> 
```
tosa_support_checks[target] = tosa_check
```

Reply:
ah, wasn't this, but a type-check. resolved in https://github.com/pytorch/executorch/pull/6918